packages/aws-cdk-lib/aws-iam/lib/principals.ts

```diff
@@ -555,6 +555,28 @@ export class ServicePrincipal extends PrincipalBase {
     return new ServicePrincipalToken(service, {}).toString();
   }
 
+  /**
+   * Return the service principal using the service principal name as it is passed to the function without
+   * any change regardless of the region used in the stack if it is Opted in or not.
+   *
+   * @example
+   * const principalName = iam.ServicePrincipal.of('elasticmapreduce.amazonaws.com.cn');
+   */
+  public static of(servicePrincipalName: string): ServicePrincipal {
```

Comment:
nit: I think the method name can be better to indicate how it is different from the existing `servicePrincipalName` method, namely, this new one use the passed in string as is. Maybe this:

```suggestion
  public static fromStaticString(servicePrincipalName: string): ServicePrincipal {
```

Reply:
done